### PR TITLE
domain: refactor schema validator interface

### PR DESCRIPTION
**Pull request by tiancaiamao**
Simplify the `SchemaValidator` interface, just the `Check` is necessary.
SchemaValidator would tell the check result: success/failure/unknown.

@zimulala @shenli

### CODE REVIEW

File: domain/schema_validator.go [hunk 1]
```diff
@@ -143,19 +149,13 @@ func (s *schemaValidator) isAllExpired(txnTS uint64) bool {
...
-	}
-
-	_, isExisting := s.itemSchemaVers[currVer]
-	if !isExisting {
-		log.Infof("the schema version %d is much older than the latest version %d", currVer, s.latestSchemaVer)
-		return false, ErrInfoSchemaChanged
+// isRelatedTablesChanged returns the result whether relatedTableIDs is changed
+// from usedVer to the latest schema version.
+// NOTE, this function should be called under lock!
+func (s *schemaValidator) isRelatedTablesChanged(txnTS uint64, currVer int64, tableIDs []int64) bool {
```
Comment (winkyao) [inline domain/schema_validator.go]: remove useless `txnTS` param?

File: domain/schema_validator.go [hunk 2]
```diff
@@ -143,19 +149,13 @@ func (s *schemaValidator) isAllExpired(txnTS uint64) bool {
 	return t.After(s.latestSchemaExpire)
 }
 
-func (s *schemaValidator) IsRelatedTablesChanged(txnTS uint64, currVer int64, tableIDs []int64) (bool, error) {
-	s.mux.RLock()
-	defer s.mux.RUnlock()
-
-	if s.isAllExpired(txnTS) {
```
Comment (winkyao) [inline domain/schema_validator.go]: why not check `isAllExpired` in other place?

Comment (tiancaiamao) [inline domain/schema_validator.go]: This check is done in `Check()` function, not `isRelatedTablesChanged()`

**Comment by tiancaiamao** [comment]
@winkyao @zimulala

**Comment by shenli** [comment]
LGTM
@zimulala @winkyao PTAL

**Comment by tiancaiamao** [comment]
@zimulala

File: session_test.go [hunk 3]
```diff
@@ -88,34 +88,34 @@ func (s *testSessionSuite) TestSchemaCheckerSimple(c *C) {
...
 	checker.schemaVer = 2
 	checker.relatedTableIDs = []int64{3}
-	err = checker.checkOnce(ts)
+	err = checker.Check(ts)
 	c.Assert(err, IsNil)
 	nowTS := uint64(time.Now().UnixNano())
-	err = checker.checkOnce(nowTS)
-	c.Assert(terror.ErrorEqual(err, domain.ErrInfoSchemaExpired), IsTrue)
+	result := checker.SchemaValidator.Check(nowTS, checker.schemaVer, checker.relatedTableIDs)
+	c.Assert(result, Equals, domain.ResultSucc)
```
Comment (zimulala) [inline session_test.go]: Why this result becomes successful

**Comment by zimulala** [comment]
LGTM
/run-all-test

**Pull request merged**